packages/web-runtime/src/components/Account/CalDavUrl.vue

```diff
@@ -0,0 +1,132 @@
+            </oc-button>
+          </div>
+        </oc-table-td>
+      </oc-table-tr>
+      <oc-table-tr class="account-page-info-caldav-username">
+        <oc-table-td>{{ $gettext('Username') }}</oc-table-td>
+        <oc-table-td colspan="2">
+          <div class="oc-flex oc-flex-middle">
+            <span>{{ user.onPremisesSamAccountName }}</span>
+            <oc-button
+              v-oc-tooltip="$gettext('Copy CalDAV username')"
+              class="oc-ml-m"
+              appearance="raw"
+              data-testid="copy-caldav-username"
+              size="small"
+              :aria-label="$gettext('Copy CalDAV username to clipboard')"
+              no-hover
+              @click="copyCalDavUsernameToClipboard"
+            >
+              <oc-icon :name="copyCalDavUsernameIcon" size="small" />
+              <span class="oc-ml-2xs">{{ $gettext('Copy') }}</span>
+            </oc-button>
+          </div>
+        </oc-table-td>
+      </oc-table-tr>
+      <oc-table-tr class="account-page-info-caldav-password">
+        <oc-table-td>{{ $gettext('Password') }}</oc-table-td>
+        <oc-table-td colspan="2">
+          {{ $gettext('An app token needs to be generated and then can be used.') }}
+        </oc-table-td>
+      </oc-table-tr>
+    </account-table>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, computed, onMounted, unref } from 'vue'
+import { storeToRefs } from 'pinia'
+import { useClientService, useConfigStore, useUserStore } from '@opencloud-eu/web-pkg'
+import { useGettext } from 'vue3-gettext'
+import AccountTable from './AccountTable.vue'
+
+const { $gettext } = useGettext()
+const userStore = useUserStore()
+const { user } = storeToRefs(userStore)
+const configStore = useConfigStore()
+const clientService = useClientService()
+
+const calDavAvailable = ref(false)
+const copiedIcon = 'check'
+const copyIcon = 'file-copy'
+const checked = ref(false)
+
+const calDavUrl = computed(() => {
+  return `${configStore.serverUrl}`
+})
+
+const copyCalDavUrlIcon = ref(copyIcon)
+const copyCalDavUsernameIcon = ref(copyIcon)
+
+const copyCalDavUrlToClipboard = () => {
+  navigator.clipboard.writeText(unref(calDavUrl))
+  copyCalDavUrlIcon.value = copiedIcon
+  setTimeout(() => (copyCalDavUrlIcon.value = copyIcon), 1500)
+}
+
+const copyCalDavUsernameToClipboard = () => {
+  navigator.clipboard.writeText(user.value.onPremisesSamAccountName)
+  copyCalDavUsernameIcon.value = copiedIcon
+  setTimeout(() => (copyCalDavUsernameIcon.value = copyIcon), 1500)
+}
+
+onMounted(async () => {
+  try {
+    const wellKnownUrl = `${configStore.serverUrl}.well-known/caldav`
+
+    try {
+      const response = await clientService.httpAuthenticated.get(wellKnownUrl, {
+        method: 'OPTIONS'
+      })
+      console.info('CalDAV check response:', response.request.responseURL)
+      if (response.request.responseURL.includes(`${configStore.serverUrl}caldav/`)) {
```

Comment:
The `web-client` package provides a helper for building urls like this. It's a bit less error prone since you don't need to manually care about the slashes.

```suggestion
      if (response.request.responseURL.includes(urlJoin(configStore.serverUrl, 'caldav')) {
```